gnssanalysis/gn_io/sp3.py

```diff
@@ -53,7 +53,7 @@
 _RE_SP3_HEAD_FDESCR = _re.compile(rb"\%c[ ]+(\w{1})[ ]+cc[ ](\w{3})")
 
 
-_SP3_DEF_PV_WIDTH = [1, 3, 14, 14, 14, 14, 1, 2, 1, 2, 1, 2, 1, 3, 1, 1, 1, 1, 1, 1]
+_SP3_DEF_PV_WIDTH = [1, 3, 14, 14, 14, 14, 1, 2, 1, 2, 1, 2, 1, 3, 1, 1, 1, 2, 1, 1]
```

Comment:
This probably isn't where the problem is arising, but seems like we are getting a misalignment of the output `P` column.

I've run a couple tests of SP3 files, reading and then writing them back out, and comparing the input and output file, there is an added space somewhere causing the `P` column to be shifted by one across:

Original IGS:
```
PG31  -6665.022843 -21628.559592  13360.498756   -226.172552  6  6  6 161       
PG32  15155.161510 -15052.638054  15871.232492   -608.029872  8  7  7 143       
*  2024  8  5  0  0  0.00000000
PG02 -22682.219867 -13324.112236   5209.362171   -385.403423  6  8  9 167  P   P
PG03 -13555.871391  -5841.138951  21922.234449    487.153854  9  6  7 207  P   P
```
After read / write cycle:
```
PG31  -6665.022843 -21628.559592  13360.498756   -226.172552  8  8  8 206        
PG32  15155.161510 -15052.638054  15871.232492   -608.029872  9  9  9 201        
*  2024  8  5  0  0  0.00000000
PG02 -22682.219867 -13324.112236   5209.362171   -385.403423  8  9 10 207   P   P
PG03 -13555.871391  -5841.138951  21922.234449    487.153854 10  8  9 216   P   P
```
An unrelated issue revealed by this is also that the STD values are being changed, but we can deal with that in another PR

Reply:
I've revised the way the `FLAGS` columns are written out, which addresses this.
Flags are now concatenated into a single string per record in the dataframe, ahead of the call to `DataFrame.to_string()`. This works around the compulsory space inserted between columns, which was causing misalignment.